docs/appendices/backend-writers-guide.rst

```diff
@@ -729,22 +729,22 @@ other update/remove functionality at a later stage.
 
 .. cpp:function:: virtual bool DNSBackend::startTransaction(const string &qname, int id)
 
-  See :cpp:func:`above <DNSBackend::beginTransaction>`. Please
+  See :cpp:func:`above <DomainInfo::startTransaction>`. Please
```

Comment:
```suggestion
  See :cpp:func:`above <DNSBackend::startTransaction>`. Please
```

Reply:
ah okay, but why does is the of the link mention "above" (we're *in* the DNSBackend class) here? pdns-authoritative-ns2

If I apply your suggestion, please note that it links to itself... :confused: 

So the whole link can be removed, imo?

Reply:
My understanding is that the "see above" link is because we are in the "DNS update support" section, and should point to the description of `::startTransaction()` in the "Read/write slave-capable backends" section. However it seems wrong there too, unless I'm mistaken, ` bool DomainInfo::isMaster()` should be `bool DNSBackend::isMaster()` and so on up to `bool DomainInfo::setFresh()` that should be `bool DNSBackend::setFresh()`.

Reply:
Yeah, I think I understand it now - quite a lot of references seem indeed wrong. I think the initial confusion by the author may be caused by the forward declaration of the DNSBackend class before the DomainInfo struct?

https://github.com/PowerDNS/pdns/blob/b4b9cf2d683845bba29933e1cb40cf917fcbadcb/pdns/dnsbackend.hh#L47-L48

But then, `isMaster()` is indeed part of the `DomainInfo` struct, not `DNSBackend`!

https://github.com/PowerDNS/pdns/blob/b4b9cf2d683845bba29933e1cb40cf917fcbadcb/pdns/dnsbackend.hh#L90-L97

So, this gets a bit too hairy to get right in this big line of commits. I suggest to take that commit out and we should open a separate PR for that document. WDYT?

Reply:
`isMaster()` is indeed part of `DomainInfo` in master since a2dfc9ea78614a9eef514a612117e507d70d4664, but not in 4.1.x, so it's going to be fun to document it properly..

Reply:
> I suggest to take that commit out and we should open a separate PR for that document. WDYT?

Makes sense to me!

Reply:
(not very fluent with GitHub workflow - I should have marked this as resolved?)
created #7650